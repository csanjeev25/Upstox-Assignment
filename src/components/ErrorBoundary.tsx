import React, { ReactNode } from 'react';

interface ErrorBoundaryProps {
  children: ReactNode;
  onError?: (error: Error, componentStack: React.ErrorInfo) => void;
  FallbackComponent?: React.ComponentType<{ resetError: () => void }>;
}

class ErrorBoundary extends React.Component {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error: Error) {
    return { hasError: true };
  }

  componentDidCatch(error: Error, info: React.ErrorInfo) {
    if (typeof this.props.onError === 'function') {
      this.props.onError(error, info.componentStack);
    }
  }

  resetError = () => {
    this.setState({ hasError: false });
  };

  render() {
    const { FallbackComponent } = this.props;
    if (this.state.hasError) {
      return FallbackComponent ? (
        <FallbackComponent resetError={this.resetError} />
      ) : null;
    }

    return this.props.children;
  }
}

export default React.memo(ErrorBoundary);
