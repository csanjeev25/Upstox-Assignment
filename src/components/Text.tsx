import React from "react"
import { StyleProp, Text as RNText, TextProps as RNTextProps, TextStyle } from "react-native"
import { colors, typography } from "../values/theme"

type Presets = keyof typeof $presets
type Sizes = keyof typeof $sizeStyles

export interface TextProps extends RNTextProps {
  /**
   * The text to display if not using `tx` or nested components.
   */
  text?: string
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<TextStyle>
  /**
   * One of the different types of text presets.
   */
  preset?: Presets
  /**
   * Children components.
   */
  children?: React.ReactNode
  /**
   * Text size modifier.
   */
  size?: Sizes
}

export function Text(props: TextProps) {
  const { size, text, children, style: $styleOverride, ...rest } = props

  const content = text || children

  const preset: Presets = props.preset ?? "default"
  const $styles: StyleProp<TextStyle> = [
    $presets[preset],
    $styleOverride,
    size && $sizeStyles[size],
  ]

  return (
    <RNText {...rest} style={$styles}>
      {content}
    </RNText>
  )
}

const $sizeStyles = {
  xxl: { fontSize: 36, lineHeight: 44 } satisfies TextStyle,
  xl: { fontSize: 24, lineHeight: 34 } satisfies TextStyle,
  lg: { fontSize: 20, lineHeight: 32 } satisfies TextStyle,
  md: { fontSize: 18, lineHeight: 26 } satisfies TextStyle,
  sm: { fontSize: 16, lineHeight: 24 } satisfies TextStyle,
  xs: { fontSize: 14, lineHeight: 21 } satisfies TextStyle,
  xxs: { fontSize: 12, lineHeight: 18 } satisfies TextStyle,
}

const $baseStyle: StyleProp<TextStyle> = [
  $sizeStyles.sm,
  {fontFamily: typography.primary.default },
  { color: colors.text },
  
]

const $presets = {
  default: $baseStyle,
  bold: [$baseStyle, { fontFamily: typography.primary.bold }] as StyleProp<TextStyle>,
  heading: [$baseStyle, $sizeStyles.xxl, typography.primary.bold, { color: colors.palette.white }] as StyleProp<TextStyle>,
  subheading: [$baseStyle, $sizeStyles.lg, typography.primary.default] as StyleProp<TextStyle>,
}
