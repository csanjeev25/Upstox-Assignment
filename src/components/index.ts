export * from "src/components/Header";
export * from "src/components/EmbryonicComponent";
export * from "src/components/ListEmptyScreen";
export * from "src/components/Text";
export * from "src/components/TouchableButton";

