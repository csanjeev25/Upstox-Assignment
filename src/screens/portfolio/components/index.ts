export * from "src/screens/portfolio/components/BreakdownItemRow";
export * from "src/screens/portfolio/components/ListRowItem";
export * from "src/screens/portfolio/components/PortfolioBreakdown";
export * from "src/screens/portfolio/components/PortfolioContainer";
export * from "src/screens/portfolio/components/PortfolioSummary";
export * from "src/screens/portfolio/components/StockItem";
export * from "src/screens/portfolio/components/StockList";
export * from "src/screens/portfolio/components/PortfolioScreen";