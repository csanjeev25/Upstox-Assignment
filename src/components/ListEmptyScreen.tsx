import React from "react"
import { View, StyleSheet, TouchableOpacity } from "react-native"
import { Text } from "./Text"
import { colors, spacing, typography } from "../values/theme"

interface IProps {
  /*
    Optional display text
  */
  displayText?: string
  /*
    on click retry
  */
  onClickHandler?: () => void
}

export function ListEmptyScreen(props: IProps) {
  const { displayText, onClickHandler } = props
  return (
    <View style={styles.container}>
      {onClickHandler ? (
        <TouchableOpacity onPress={onClickHandler}>
          <Text style={styles.displayText} text={displayText} size="md"></Text>
          <Text style={styles.button} size="lg" text="Retry"></Text>
        </TouchableOpacity>
      ) : (
        <Text style={styles.displayText} text={displayText} size="md"></Text>
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  displayText: {
    color: colors.palette.greyish,
    fontFamily: typography.primary.normal,
    marginVertical: spacing.grid_2,
  },
  button: { color: colors.palette.seafoamBlue, fontFamily: typography.primary.semibold },
})
