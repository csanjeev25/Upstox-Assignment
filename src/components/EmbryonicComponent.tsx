import React, { SetStateAction } from "react";
import { ActivityIndicator, StyleSheet, View } from "react-native";
import { colors, spacing, strings } from "src/values";
import { ListEmptyScreen } from "src/components";
import { testIdProps } from "src/utils";

export interface EmbryonicComponentProps {
  /**
   * Optional to retry on network
   */
  setConnectivityToggle?: SetStateAction<boolean>
  /**
   * Optional check if loading
   */
  isLoading?: boolean
  /**
   * data to display
   */
  data?: object | Array<any> | undefined | null
  /**
   * Optional check if internet is connected.
   */
  hasInternetConnection?: boolean | null
  /**
   * Optional check if any error.
   */
  hasError?: boolean
  /**
   * Override try again operation.
   */
  tryAgain?: () => void
}

export const EmbryonicComponent = ({
  setConnectivityToggle,
  isLoading,
  data,
  hasInternetConnection,
  hasError,
  tryAgain,
}: EmbryonicComponentProps) => {
  const tryAgainHandler = () => {
    typeof tryAgain === "function" ? tryAgain() : setConnectivityToggle?.((prev) => !prev)
  }

  if (isLoading) {
    return (
        <View style={styles.container} {...testIdProps("loading-component")}>
          <ActivityIndicator style={styles.container} size="large" color='purple' />
        </View>
      )
  }

  if (!data && !hasInternetConnection) {
    return (
      <ListEmptyScreen
        displayText={strings.noInternetConnection}
        onClickHandler={tryAgainHandler}
      />
    )
  }

  if (hasError) {
    return (
      <ListEmptyScreen displayText={strings.somethingWentWrong} onClickHandler={tryAgainHandler} {...testIdProps("list-empty-component")}/>
    )
  }

  return null
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: "100%",
    width: "100%",
    zIndex: 3,
    backgroundColor: colors.palette.white,
  },
  scrollableContainer: {
    flexGrow: 1,
    paddingBottom: spacing.grid_13_half,
  },
})
