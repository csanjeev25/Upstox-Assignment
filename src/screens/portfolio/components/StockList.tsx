import React from "react"
import { StyleSheet, Text, View } from "react-native"
import StockItem from "./StockItem"
import { colors, spacing } from "../../../values/theme/"
import { IStockList, UserHoldings } from "../types"
import ListEmptyScreen from "../../../components/ListEmptyScreen"
import { isNullOrEmptyArray } from "../../../utils/common.utils"
import { ScrollView } from "react-native-gesture-handler"

interface IRenderItem {
  item: UserHoldings | null | undefined
  index: number
}

function StockList(props: IStockList) {
  const { stockData, reload } = props

  const rowRenderer = ({ item, index }: IRenderItem) => {
    if (!item) {
      return null
    }
    return (
      <View key={item?.symbol}>
        <StockItem stockData={item} index={index} />
        {renderSeparator(index)}
      </View>
    )
  }

  const renderSeparator = (index) => {
    if (index < stockData.length - 1) {
      return (
        <View style={styles.separator} />
      )
    }
  }

  return (
    <View style={styles.containerStyle}>
      {isNullOrEmptyArray(stockData) ? (
        <ListEmptyScreen onClickHandler={reload} />
      ) : (
        <ScrollView contentContainerStyle={styles.container}>
          {stockData?.map((item, index) => (
            rowRenderer({item, index})
          ))}
        </ScrollView>
      )}
    </View>
  )
}

export default StockList

const styles = StyleSheet.create({
  containerStyle: {
    backgroundColor: colors.palette.white,
  },
  separator: {
    height: 1,
    backgroundColor: colors.palette.whiteFive,
    marginLeft: spacing.grid_5_half,
    marginRight: spacing.grid_5_half,
  },
  container: {
    width: '100%',
    flexDirection: 'column',
  }
})
