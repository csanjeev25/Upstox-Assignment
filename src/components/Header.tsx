import React from "react"
import { View, StyleSheet, StyleProp, ViewStyle, TextStyle, TextProps } from "react-native"
import { Text } from "../components/Text"
import { ExtendedEdge, useSafeAreaInsetsStyle } from "../hooks/useSafeAreaInsetsStyle"
import { colors, fontSize, spacing, typography } from "../values/theme"
import { IosUtils } from "../utils/ios.utils";

export interface HeaderProps {
  /**
   * Optional title style override.
   */
  titleStyle?: StyleProp<TextStyle>
  /**
   * Optional inner header wrapper style override.
   */
  style?: StyleProp<ViewStyle>
  /**
   * Title text to display
   */
  title?: TextProps["text"]
  /**
   * Optional inner header wrapper style override.
   */
  containerStyleOverride?: StyleProp<ViewStyle>
  /**
   * Optional inner header wrapper style override.
   */
  titleStyleOverride?: StyleProp<ViewStyle>
  /**
   * Override the default edges for the safe area.
   */
  safeAreaEdges?: ExtendedEdge[]
}

export function Header(props: HeaderProps) {
  const {
    title,
    titleStyle,
    titleStyleOverride,
    containerStyleOverride,
    safeAreaEdges,
  } = props
  const $containerInsets = useSafeAreaInsetsStyle(safeAreaEdges)
  return (
    <View style={[$containerInsets, containerStyleOverride ? containerStyleOverride : styles.container]}>
      <Text
        size="md"
        text={title}
        style={[titleStyleOverride ? titleStyleOverride : styles.title, titleStyle]}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: 'purple',
    height: spacing.grid_7,
    elevation: 2,
    ...IosUtils.getIosElevation(spacing.xxxs),
  },
  title: {
    fontSize: fontSize.h2,
    color: colors.palette.white,
    fontFamily: typography.primary?.bold,
    marginLeft: spacing.grid_2,
  },
})
