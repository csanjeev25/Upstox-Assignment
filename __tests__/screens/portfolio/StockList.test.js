import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { StockList } from '../../../src/screens/portfolio/components/StockList';

jest.mock('../../../src/screens/portfolio/helper/checks', () => ({
  checkIfValidStock: jest.fn(),
}));

jest.mock('../../../src/utils/common.utils', () => ({
  isNullOrEmptyArray: jest.fn(),
}));

jest.mock('src/components/ListEmptyScreen', () => 'ListEmptyScreen');

const mockStockData = [
  {
    symbol: "TCS",
    quantity: 10,
    ltp: 3250.5,
    avgPrice: 2480.3,
    close: 3312,
  },
  {
    symbol: "Wipro",
    quantity: 80,
    ltp: 550.2,
    avgPrice: 380.3,
    close: 580,
  },
  {
    symbol: "SBI",
    quantity: 12,
    ltp: 650.5,
    avgPrice: 680.3,
    close: 613,
  },
  {
    symbol: "TataMotors",
    quantity: 100,
    ltp: 650.5,
    avgPrice: 280.3,
    close: 780,
  },
  {
    symbol: "Reliance",
    quantity: 10,
    ltp: 2887.1,
    avgPrice: 2780.3,
    close: 2610,
  },
]

describe('<StockList />', () => {
  it('renders correctly with stock data', () => {
    const { getByText } = render(<StockList stockData={mockStockData} reload={() => {}} />);
    expect(getByText('TCS')).toBeTruthy();
    expect(getByText('Wipro')).toBeTruthy();
  });

  it('renders the empty state when no stock data is available', () => {
    const { getAllByTestId } = render(<StockList stockData={[]} reload={() => {}} />);
    expect(getAllByTestId('list-empty-component')).toBeTruthy();
  });

  it('matches snapshot with stock data', () => {
    const tree = render(<StockList stockData={mockStockData} reload={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when empty', () => {
    const tree = render(<StockList stockData={[]} reload={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});