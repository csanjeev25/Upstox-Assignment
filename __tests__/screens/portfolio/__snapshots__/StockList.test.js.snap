// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StockList /> matches snapshot when empty 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <RCTScrollView
    accessibilityLabel="stock-list-scroll"
    contentContainerStyle={
      {
        "flexDirection": "column",
        "width": "100%",
      }
    }
    testID="stock-list-scroll"
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<StockList /> matches snapshot with stock data 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <RCTScrollView
    accessibilityLabel="stock-list-scroll"
    contentContainerStyle={
      {
        "flexDirection": "column",
        "width": "100%",
      }
    }
    testID="stock-list-scroll"
  >
    <View>
      <View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                [
                  [
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "Lato-Regular",
                    },
                    {
                      "color": "#484848",
                    },
                  ],
                  {
                    "fontFamily": "Lato-Bold",
                  },
                  undefined,
                ]
              }
            >
              TCS
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                LTP: 
              </Text>
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    {
                      "fontFamily": "Lato-Bold",
                    },
                    undefined,
                  ]
                }
              >
                ₹ 3250.50
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                [
                  [
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "Lato-Regular",
                    },
                    {
                      "color": "#484848",
                    },
                  ],
                  undefined,
                  undefined,
                ]
              }
            >
              10
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                P/L: 
              </Text>
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    {
                      "fontFamily": "Lato-Bold",
                    },
                    undefined,
                  ]
                }
              >
                ₹ 7702.00
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#efefef",
              "height": 1,
              "marginLeft": 20,
              "marginRight": 20,
            }
          }
        />
      </View>
      <View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                [
                  [
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "Lato-Regular",
                    },
                    {
                      "color": "#484848",
                    },
                  ],
                  {
                    "fontFamily": "Lato-Bold",
                  },
                  undefined,
                ]
              }
            >
              Wipro
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                LTP: 
              </Text>
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    {
                      "fontFamily": "Lato-Bold",
                    },
                    undefined,
                  ]
                }
              >
                ₹ 550.20
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                [
                  [
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "Lato-Regular",
                    },
                    {
                      "color": "#484848",
                    },
                  ],
                  undefined,
                  undefined,
                ]
              }
            >
              80
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                P/L: 
              </Text>
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    {
                      "fontFamily": "Lato-Bold",
                    },
                    undefined,
                  ]
                }
              >
                ₹ 13592.00
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#efefef",
              "height": 1,
              "marginLeft": 20,
              "marginRight": 20,
            }
          }
        />
      </View>
      <View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                [
                  [
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "Lato-Regular",
                    },
                    {
                      "color": "#484848",
                    },
                  ],
                  {
                    "fontFamily": "Lato-Bold",
                  },
                  undefined,
                ]
              }
            >
              SBI
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                LTP: 
              </Text>
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    {
                      "fontFamily": "Lato-Bold",
                    },
                    undefined,
                  ]
                }
              >
                ₹ 650.50
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                [
                  [
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "Lato-Regular",
                    },
                    {
                      "color": "#484848",
                    },
                  ],
                  undefined,
                  undefined,
                ]
              }
            >
              12
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                P/L: 
              </Text>
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    {
                      "fontFamily": "Lato-Bold",
                    },
                    undefined,
                  ]
                }
              >
                ₹ -357.60
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#efefef",
              "height": 1,
              "marginLeft": 20,
              "marginRight": 20,
            }
          }
        />
      </View>
      <View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                [
                  [
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "Lato-Regular",
                    },
                    {
                      "color": "#484848",
                    },
                  ],
                  {
                    "fontFamily": "Lato-Bold",
                  },
                  undefined,
                ]
              }
            >
              TataMotors
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                LTP: 
              </Text>
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    {
                      "fontFamily": "Lato-Bold",
                    },
                    undefined,
                  ]
                }
              >
                ₹ 650.50
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                [
                  [
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "Lato-Regular",
                    },
                    {
                      "color": "#484848",
                    },
                  ],
                  undefined,
                  undefined,
                ]
              }
            >
              100
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                P/L: 
              </Text>
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    {
                      "fontFamily": "Lato-Bold",
                    },
                    undefined,
                  ]
                }
              >
                ₹ 37020.00
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#efefef",
              "height": 1,
              "marginLeft": 20,
              "marginRight": 20,
            }
          }
        />
      </View>
      <View>
        <View
          style={
            {
              "backgroundColor": "#FFFFFF",
              "paddingHorizontal": 16,
              "paddingVertical": 12,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                [
                  [
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "Lato-Regular",
                    },
                    {
                      "color": "#484848",
                    },
                  ],
                  {
                    "fontFamily": "Lato-Bold",
                  },
                  undefined,
                ]
              }
            >
              Reliance
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                LTP: 
              </Text>
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    {
                      "fontFamily": "Lato-Bold",
                    },
                    undefined,
                  ]
                }
              >
                ₹ 2887.10
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                [
                  [
                    {
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                    {
                      "fontFamily": "Lato-Regular",
                    },
                    {
                      "color": "#484848",
                    },
                  ],
                  undefined,
                  undefined,
                ]
              }
            >
              10
            </Text>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                P/L: 
              </Text>
              <Text
                style={
                  [
                    [
                      {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      {
                        "fontFamily": "Lato-Regular",
                      },
                      {
                        "color": "#484848",
                      },
                    ],
                    {
                      "fontFamily": "Lato-Bold",
                    },
                    undefined,
                  ]
                }
              >
                ₹ 1068.00
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
